geomesa-spark-sql/geomesa-spark-sql-core/src/main/scala/org/locationtech/geomesa/spark/GeoMesaSparkSQL.scala

```diff
@@ -0,0 +1,270 @@
+// {{
+// val df = spark.read
+//   .format("geomesa")
+//   .option(GM.instanceIdParam.getName, "mycloud")
+//   .option(GM.userParam.getName, "user")
+//   .option(GM.passwordParam.getName, "password")
+//   .option(GM.tableNameParam.getName, "sparksql")
+//   .option(GM.mockParam.getName, "true")
+//   .option("geomesa.feature", "chicago")
+//   .load()
+// }}
+class GeoMesaDataSource extends DataSourceRegister with RelationProvider with SchemaRelationProvider with CreatableRelationProvider {
+
+  override def shortName(): String = "geomesa"
+
+  override def createRelation(sqlContext: SQLContext, parameters: Map[String, String]): BaseRelation = {
+    SQLTypes.init(sqlContext)
+    val ds = DataStoreFinder.getDataStore(parameters)
+    val sft = ds.getSchema(parameters("geomesa.feature"))
+    val schema = sft2StructType(sft)
+    GeoMesaRelation(sqlContext, sft, schema, parameters)
+  }
+
+  // JNH: Q: Why doesn't this method have the call to SQLTypes.init(sqlContext)?
+  override def createRelation(sqlContext: SQLContext, parameters: Map[String, String], schema: StructType): BaseRelation = {
+    val ds = DataStoreFinder.getDataStore(parameters)
+    val sft = ds.getSchema(parameters("geomesa.feature"))
+    GeoMesaRelation(sqlContext, sft, schema, parameters)
+  }
+
+  private def sft2StructType(sft: SimpleFeatureType) = {
+    val fields = sft.getAttributeDescriptors.map { ad => ad2field(ad) }.toList
+    StructType(StructField("__fid__", DataTypes.StringType, nullable =false) :: fields)
+  }
+
+  def structType2SFT(struct: StructType, name: String): SimpleFeatureType = {
+    import java.{lang => jl}
+    val fields = struct.fields
+
+    val builder = new SimpleFeatureTypeBuilder
+
+    fields.filter( _.name != "__fid__").foreach {
+      field =>
+        field.dataType match {
+          case DataTypes.BooleanType => builder.add(field.name, classOf[jl.Boolean])
+          case DataTypes.DateType => builder.add(field.name, classOf[java.util.Date])
+          case DataTypes.FloatType => builder.add(field.name, classOf[jl.Float])
+          case DataTypes.IntegerType => builder.add(field.name, classOf[jl.Integer])
+          case DataTypes.DoubleType => builder.add(field.name, classOf[jl.Double])
+          case DataTypes.StringType => builder.add(field.name, classOf[jl.String])
+          case DataTypes.LongType   => builder.add(field.name, classOf[jl.Long])
+
+          case SQLTypes.PointType => builder.add(field.name, classOf[com.vividsolutions.jts.geom.Point])
+          case SQLTypes.LineStringType => builder.add(field.name, classOf[com.vividsolutions.jts.geom.LineString])
+          case SQLTypes.PolygonType  => builder.add(field.name, classOf[com.vividsolutions.jts.geom.Polygon])
+          case SQLTypes.MultipolygonType  => builder.add(field.name, classOf[com.vividsolutions.jts.geom.MultiPolygon])
+          case SQLTypes.GeometryType => builder.add(field.name, classOf[com.vividsolutions.jts.geom.Geometry])
```

Comment:
not quite sure how this gets used - sounds like the sft should already exist? if you are creating a schema from this, you will need the 'override.geom' flag